firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Util/COBS.tpp

```
@@ -31,13 +31,19 @@ namespace Pufferfish::Util {
 
 static const size_t max_block_size = 254;
 
-size_t encode_cobs(const uint8_t *buffer, size_t size, uint8_t *encoded_buffer) {
+template <size_t input_size, size_t output_size>
+IndexStatus encode_cobs(
+    const Util::ByteVector<input_size> &buffer, Util::ByteVector<output_size> &encoded_buffer) {
   size_t read_index = 0;
   size_t write_index = 1;
   size_t code_index = 0;
   uint8_t code = 1;
 
-  while (read_index < size) {
+  if (encoded_buffer.empty()) {
+    encoded_buffer.resize(buffer.size() + 1);
```

Shouldn't we always do this resize? And shouldn't we use `get_cobs_encoded_buffer_size` to tell us the size to resize to?

We should return an error code if the resize fails and returns an error.